src/python/bcc/__init__.py

```diff
@@ -768,15 +766,15 @@ def kprobe_poll(self, timeout = -1):
             exit()
 
     def cleanup(self):
-        for k, v in self.open_kprobes.items():
-            lib.perf_reader_free(v)
-            # non-string keys here include the perf_events reader
-            if isinstance(k, str):
+        for k in list(self.open_kprobes):
+            lib.perf_reader_free(self.open_kprobes[k])
+            # perf buffers have tuple keys and don't need detaching
+            if not isinstance(k, tuple):
                 desc = "-:kprobes/%s" % k
                 lib.bpf_detach_kprobe(desc.encode("ascii"))
             self._del_kprobe(k)
-        for k, v in self.open_uprobes.items():
-            lib.perf_reader_free(v)
+        for k in list(self.open_uprobes):
```

Comment:
Actually, you could also have done:
```
for k, v in list(self.open_uprobes.items()):
```
Which would have avoided the `[k]` below.